src/components/CRUDDash.tsx

```diff
@@ -0,0 +1,288 @@
+  app: FirebaseApp
+}
+
+const CRUDDash: React.FC<CrudProps> = ({ app }) => {
+  const firestore = getFirestore(app)
+
+  const [entities, setEntities] = useState<string[]>([])
+  const [isLoading, setIsLoading] = useState(false)
+  const [showAddEntity, setShowAddEntity] = useState(false)
+  const [addEntityName, setAddEntityName] = useState('')
+  const [addEntitySecondProperty, setAddEntitySecondProperty] = useState('')
+  const [showModifyEntity, setShowModifyEntity] = useState(false)
+  const [modifyEntityOldName, setModifyEntityOldName] = useState('')
+  const [modifyEntityNewName, setModifyEntityNewName] = useState('')
+  const [modifyEntitySecondProperty, setModifyEntitySecondProperty] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+  const [entityType, setEntityType] = useState<'Categories' | 'Types'>('Categories')
+
+  const showEntities = async () => {
+    setIsLoading(true)
+    try {
+      const q = collection(firestore, entityType)
+      const querySnapshot = await getDocs(q)
+      const entitiesArray: string[] = []
+      querySnapshot.forEach((doc) => {
+        const entity = doc.data()
+        entitiesArray.push(entity.name)
+      })
+      setEntities(entitiesArray)
+    } catch (error) {
+      setErrorMessage('Error al obtener ' + entityType.toLowerCase())
+      window.alert(errorMessage)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  const handleAddEntityNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setAddEntityName(event.target.value)
+  }
+
+  const handleAddEntityColorChange = (color: any) => {
+    setAddEntitySecondProperty(color.hex) // Actualiza el valor del color
+  }
+
+  const handleModifyEntityColorChange = (color: any) => {
+    setModifyEntitySecondProperty(color.hex)
+  }
+
+  const handleAddEntity = async () => {
+    try {
+      if (addEntityName.trim() === '' || addEntitySecondProperty.trim() === '') {
+        return // No hagas nada si alguno de los campos de entrada está vacío
+      }
+
+      let secondProperty = addEntitySecondProperty
+
+      if (entityType === 'Types') {
+        // If entityType is 'Types', search for the category name in Firestore
+        const categoryQuery = query(collection(firestore, 'Categories'), where('name', '==', addEntitySecondProperty))
+        const categorySnapshot = await getDocs(categoryQuery)
+
+        if (!categorySnapshot.empty) {
+          // If category found, use its document ID as the category ID for 'Types'
+          categorySnapshot.forEach((doc) => {
+            secondProperty = doc.id
+          })
+        } else {
+          // Handle case when category not found
+          setErrorMessage('Categoría no encontrada')
+          window.alert(errorMessage)
+          return
+        }
+      }
+
+      // Add the entity with the appropriate second property
+      await addDoc(collection(firestore, entityType), {
+        name: addEntityName,
+        categoryID: secondProperty,
```

Comment:
This logic makes the color go in the `categoryID` property 💀
<img width="501" alt="Screen Shot 2024-04-14 at 3 49 55 PM" src="https://github.com/DSSD-Madison/Red-CORAL/assets/78292128/3c9be498-b146-4478-8c63-f3c3c43dff7f">